dvc/pkg/base.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import unicode_literals
+
+import os
+import copy
+import shutil
+
+from dvc.utils.compat import urlparse, str, builtin_str
+
+from dvc.config import Config
+from dvc.remote.config import remote_add
+
+from .config import get_pkg_settings
+
+
+class PkgBASE(object):
+    PKG_DIR = "pkg"
+
+    def __init__(self, config, settings):
+        self.settings = copy.copy(settings)
+
+        name = None
+        if isinstance(settings, str) or isinstance(settings, builtin_str):
```

Comment:
yeah, looks strange ... why not use **kwargs always ... force clients specify `name=...` always

Reply:
or `url=` instead

Reply:
Pkg might look like:
```
pkg: mypkg
```
or
```
pkg: https://github.com/iterative/mypkg
```
or
```
pkg:
  url: https://github.com/iterative/mypkg
  revision: testing
```
or
```
pkg:
  name: mypkg
  revision: testing
```
So kwargs don't work here that well.